mythril_core/src/vm.rs

```diff
@@ -16,6 +16,9 @@ use alloc::sync::Arc;
 use alloc::vec::Vec;
 use spin::RwLock;
 
+pub type ImageMapper =
+    dyn Fn(&[u8], &GuestPhysAddr, &mut GuestAddressSpace) -> Result<()>;
+
```

Comment:
Not even a comment? For shame! But seriously, I'm not clear what the idea is here? Is this trying to express that it is a function that maps some collection of bytes to a location in the guest address space? It seems like the implementation on `VirtualMachine` in `map_data` is pretty much the only reasonable way to do that. I'd rather just make that a method on `GuestAddressSpace`.

Reply:
😆  yeah I should have added a comment. I think a method on the `GuestAddressSpace` would be better. I'll add that in the update.